custom_components/tuya_local/devices/atorch_s1wp.yaml

```diff
@@ -0,0 +1,575 @@
+    dps:
+      - id: 104
+        type: integer
+        name: value
+        unit: V
+        range:
+          min: 1
+          max: 2750
+        mapping:
+          - scale: 10
+            step: 1
+  - entity: number
+    category: config
+    name: Overcurrent value
+    icon: "mdi:flash-triangle"
+    dps:
+      - id: 105
+        type: integer
+        name: value
+        unit: A
+        range:
+          min: 1
+          max: 2000
+        mapping:
+          - scale: 100
+            step: 1
+  - entity: number
+    category: config
+    name: Overpower value
+    icon: "mdi:flash-triangle"
+    dps:
+      - id: 106
+        type: integer
+        name: value
+        unit: W
+        range:
+          min: 1
+          max: 4500
+        mapping:
+          - scale: 1
+            step: 1
+  - entity: select
+    category: config
+    icon: "mdi:web-box"
+    name: Language
+    dps:
+      - id: 107
+        type: string
+        name: option
+        mapping:
+          - dps_val: chinese
+            value: Chinese
+          - dps_val: english
+            value: English
+  - entity: number
+    category: config
+    name: Screen brightness
+    icon: "mdi:brightness-7"
+    dps:
+      - id: 108
+        type: integer
+        name: value
+        range:
+          min: 1
+          max: 9
+        mapping:
+          - scale: 1
+            step: 1
+  - entity: number
+    category: config
+    name: Standby brightness
+    icon: "mdi:brightness-4"
+    dps:
+      - id: 109
+        type: integer
+        name: value
+        range:
+          min: 1
+          max: 9
+        mapping:
+          - scale: 1
+            step: 1
+  - entity: number
+    category: config
+    name: Standby time
+    icon: "mdi:sun-clock"
+    dps:
+      - id: 110
+        type: integer
+        name: value
+        unit: s
+        range:
+          min: 3
+          max: 99
+        mapping:
+          - scale: 1
+            step: 1
+  - entity: switch
+    name: Beep
+    category: config
+    dps:
+      - id: 111
+        type: boolean
+        name: switch
+  - entity: select
+    name: Switch mode
+    category: config
+    dps:
+      - id: 112
+        type: string
+        name: option
+        mapping:
+          - dps_val: controlled
+            value: Controlled
+          - dps_val: normally_open
+            value: Normally Open
+  - entity: button
+    name: Reset data
+    category: diagnostic
+    icon: "mdi:autorenew"
+    dps:
+      - id: 113
+        type: boolean
+        name: button
+        optional: true
+  - entity: button
+    name: Reset Wi-Fi
+    category: diagnostic
+    icon: "mdi:autorenew"
+    dps:
+      - id: 114
+        type: boolean
+        name: button
+        optional: true
+  - entity: button
+    name: Factory reset
+    category: diagnostic
+    icon: "mdi:autorenew"
+    dps:
+      - id: 115
+        type: boolean
+        name: button
+        optional: true
+  - entity: button
```

Comment:
Is button correct? Do you send another "true" to rotate it back?

Reply:
yes, it triggers the action in the plug, it doesn't have states